druntime/src/object.d

```diff
@@ -1665,6 +1665,7 @@ class TypeInfo_Class : TypeInfo
     void*       deallocator;
     OffsetTypeInfo[] m_offTi;
     void function(Object) defaultConstructor;   // default Constructor
+    ulong[2] nameSig;            /// unique signature for `name`
 
     immutable(void)* m_RTInfo;        // data for precise GC
```

Comment:
New fields should only ever be added at the end for strict abi compatibility. Might be a good idea to add a new class flag too `hasNameSig = 0x200`.

Reply:
I moved it to the end.

Reply:
The whole reason for nameSig is performance. Having to test a flag for it won't kill the performance, but it will degrade it.

Reply:
> The whole reason for nameSig is performance. Having to test a flag for it won't kill the performance, but it will degrade it.

It'll still be better than string comparison, but it at least gives a migration/fallback in case someone forgot to recompile all their code. Shouldn't need to keep it in for too many releases.

Reply:
Though I won't be the one explaining to someone why their project crashed on the NG. You've already explained the risk in the changelog, I'll leave it to you whether we do all we can to avoid it altogether.

Reply:
Ack, I had to move namSig back into its original position because the last field is of variable length.

Reply:
ok, I put the flag test in for now

Reply:
As for the last field moving, it's only supposed to be accessed through the member function.

Reply:
Can that function do something sneaky such that it won't link if linked with the wrong ABI? Could one inject a symbol reference after the return statement or something devious like that?

Reply:
Either way I think the risk of this is very low, (say) dub will rebuild per compiler version

Reply:
I had no idea that last field was variadic - I was under the impression that it was just 0x0 or 0x1

Reply:
I've checked this, and it's a pointer to a symbol, not variadic. So there's no blocking reason to have it at the end.